File: engine/b/src/main/java/net/crazyinvestor/engine_b/repository/ExchangeRepository.java
package net.crazyinvestor.engine_b.repository;

import net.crazyinvestor.engine_b.entity.Exchange;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface ExchangeRepository extends JpaRepository<Exchange, Long> {
    Optional<Exchange> findByName(String name);
}

File: engine/a/ticker_producer_engine/src/main/kotlin/net/crazyinvestor/ticker_producer_engine/deprecated/repository/ExchangeRepository.kt
package net.crazyinvestor.ticker_producer_engine.deprecated.repository;

import net.crazyinvestor.ticker_producer_engine.entity.Exchange
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
class ExchangeRepository {
    fun findByName(name: String): Optional<Exchange> {
        println("MOCK REPOSITORY1 $name")
        return Optional.of(Exchange(null, name))
    }
    fun save(exchange: Exchange): Exchange {
        println("save() MOCK REPOSITORY1 $exchange")
        return exchange
    }
}
